src/apps/pillarx-app/index.tsx

```diff
@@ -1,39 +1,75 @@
+import { useEffect, useState } from 'react';
 import styled from 'styled-components';
 import { useTranslation } from 'react-i18next';
 
+// types
+import { Projection } from '../../types/api';
+
 // hooks
-import { useApiData } from './hooks/useApiData';
+import { useGetTilesInfoQuery } from './api/apiSlice';
+import { useWalletAddress } from '@etherspot/transaction-kit';
 
 // components
 import PillarXLogo from './components/PillarXLogo/PillarXLogo';
 import pillarLogoLight from './images/pillarX_full_white.png';
 import H1 from './components/Typography/H1';
-import PortfolioOverview from './components/PortfolioOverview/PortfolioOverview';
-import TokensHorizontalTile from './components/TokensHorizontalTile/TokensHorizontalTile';
-import TokensVerticalTile from './components/TokensVerticalTile/TokensVerticalTile';
+import { componentMap } from './utils/configComponent';
+import SkeletonTiles from './components/SkeletonTile/SkeletonTile';
 
 export const App = () => {
   const [t] = useTranslation();
+  const [page, setPage] = useState(1);
+  const [pageData, setPageData] = useState<Projection[]>([]);
+
+  const walletAddress = useWalletAddress();
+  const { data: apiData, isLoading: isApiLoading, isFetching } = useGetTilesInfoQuery( { page: page, address: walletAddress || '' });
+
+  useEffect(() => {
+    // when apiData loads, we save it in a state to keep previous data
+      if (apiData) {
```

Comment:
May be worth importing `isSuccess` from the `useGetTilesInfoQuery` hook and adding it into the `if` statement here. This ensures that the request was successful before calling `setPageData` below otherwise it may load the cached data and we'll see the same tiles loading again

Reply:
Yes that's a good point, I've added it in, please check